### PR TITLE
Initial API specification.  

### PR DESCRIPTION
**Pull request by ryan-endacott**
Hi everyone, before I start implementing the API, I wanted to make sure we all agree on the endpoints.  This is just an initial suggestion of what they could look like.  There are probably edge cases and other things I haven't thought of yet, but let me know what you think of the endpoints.  

I want to hear from everyone before merging.  

@EaseApp/hackers 

Note:  I haven't added any web socket endpoints yet.

### CODE REVIEW

**Comment by ryan-endacott** [comment]
These could also be subject to change if we find out a better way after implementing.

File: README.md [hunk 1]
```diff
@@ -23,7 +23,38 @@ If you wish to have the server restart on file change, run the server with the `
...
+#### Go say hello
 
 Visit `localhost:3000` to see the hello world.
 
+
+## API Specification
+
+#### Login Methods
+
+TODO - I'm not sure how oauth with a frontend framework will work.
```
Comment (ryan-endacott) [inline README.md]: @karrde00 - thoughts?  Will it be the same as normal oauth, then we'll pass a token to the frontend?

**Comment by DrewWeth** [comment]
The developer of the app will know the his/her application endpoints (/users/, /itemsOnSale/,etc), they should then pass an object to search by. Looks good.

**Comment by Felttrip** [comment]
Makes sense to me. :+1:

**Comment by ryan-endacott** [comment]
Closing this because the endpoints are done.

**Pull request closed without merging**